### PR TITLE
ci: revisit OpenBSD/FreeBSD config

### PR DESCRIPTION
**Pull request by blueyed**
Brings it in line with the FreeBSD one, and splits it into more separate tasks.

### CODE REVIEW

File: .builds/freebsd.yml [hunk 1]
```diff
@@ -35,9 +35,6 @@ tasks:
 - unittest: |
     cd neovim
     gmake unittest
-- oldtest: |
-    cd neovim
-    gmake oldtest
```
Comment (blueyed) [inline .builds/freebsd.yml]: I've ran it manually from a build with a shell, and this is the result:
```
Executed:  1437 Tests
 Skipped:    52 Tests
  FAILED:     2 Tests


Failures:
        From test_writefile.vim:
        Found errors in Test_writefile_fails_conversion():
        function RunTheTest[37]..Test_writefile_fails_conversion line 12: command did not fail: write ++enc=cp932
        function RunTheTest[37]..Test_writefile_fails_conversion line 13: Expected ['line one', 'line two'] but got ['first line', 'cannot convert ', 'third line']
        From test_alot.vim:
        Found errors in Test_libcall_libcallnr():
        Caught exception in Test_libcall_libcallnr(): Vim(call):dlerror = "/usr/local/lib/: read error: Invalid argument" @ function RunTheTest[37]..Test_libcall_libcallnr, line 32

TEST FAILURE
gmake[1]: *** [Makefile:94: report] Error 1
gmake[1]: Leaving directory '/usr/home/build/neovim/src/nvim/testdir'
gmake: *** [Makefile:119: oldtest] Error 2
[build@build /usr/home/build/neovim]$ uname -a
FreeBSD build 12.1-RELEASE-p1 FreeBSD 12.1-RELEASE-p1 GENERIC  amd64
```

For some reason (Sourcehut bug?) it does not generate a log for it apparently still though.

File: .builds/openbsd.yml [hunk 2]
```diff
@@ -16,26 +16,28 @@ packages:
...
-    mkdir .deps
-    cd .deps
+    mkdir neovim/.deps
+    cd neovim/.deps
     cmake -G Ninja ../third-party/
     cmake --build . --config Debug
-    cd ..
-    mkdir build
-    cd build
-    cmake -G Ninja -DMIN_LOG_LEVEL=3 ..
```
Comment (justinmk) [inline .builds/openbsd.yml]: `MIN_LOG_LEVEL=3` is intentional and should be used in all CI jobs (faster (avoids disk) and less noisy for the `NVIM_LOG_FILE` tail). 

We might change to `MIN_LOG_LEVEL=2` after we sort out the warning-level messages. https://github.com/neovim/neovim/pull/8917

**Comment by justinmk** [comment]
Why switch to gmake? We should move in the other direction, not increase dependency on make. It's good to have living examples of using CMake directly..

**Comment by blueyed** [comment]
> Why switch to gmake? We should move in the other direction, not increase dependency on make. It's good to have living examples of using CMake directly..

ok, will revisit/revert then.
Thought it would be better for consistency - here e.g. `gmake oldtest` then triggered a rebuild.

**Comment by justinmk** [comment]
> Thought it would be better for consistency - here e.g. `gmake oldtest` then triggered a rebuild.

Hmm. Does that indicate a bug in the cmake target?

**Review by justinmk — APPROVED** [review]

**Comment by blueyed** [comment]
> Hmm. Does that indicate a bug in the cmake target?

No, it was due to different cmake options being used then (out Makefile used the exported ones while the direct cmake did not), at least that's what I've thought.

**Review by justinmk — APPROVED** [review]

**Pull request merged**